play/src/pages/GameInProgress.tsx

```diff
@@ -144,41 +172,83 @@ export default function GameInProgress({
 
   const handleSubmitAnswer = async (answerText: string) => {
     try {
-      await apiClient.addTeamAnswer(
+      const response = await apiClient.addTeamAnswer(
         teamMemberId,
         currentQuestion.id,
         answerText,
         currentState === GameSessionState.CHOOSE_CORRECT_ANSWER,
         currentState !== GameSessionState.CHOOSE_CORRECT_ANSWER
       );
+      setTeamAnswerId(response.id);
       setSelectSubmitAnswer((prev) => ({ ...prev, isSubmitted: true }));
       setDisplaySubmitted(true);
     } catch {
-      setIsError(true);
+      setIsAnswerError(true);
     }
   };
 
   const handleRetry = () => {
-    setIsError(false);
-    setSelectSubmitAnswer((prev) => ({ ...prev, isSubmitted: false }));
+    if (isAnswerError) {
+      setIsAnswerError(false);
+      setSelectSubmitAnswer((prev) => ({ ...prev, isSubmitted: false }));
+    }
+    if (isConfidenceError) {
+      setIsConfidenceError(false);
+      setSelectConfidence({
+        timeOfLastSelect: null,
+        selectedConfidenceIndex: null,
+        isSelected: false,
+      });
+    }
   };
 
   const handleSelectAnswer = (index: number) => {
     setSelectSubmitAnswer((prev) => ({ ...prev, selectedAnswerIndex: index }));
   };
 
+  const setTimeOfLastConfidenceSelect = (time: number | null) => {
+    setSelectConfidence((prev) => ({ ...prev, timeOfLastSelect: time }));
+  };
+
+  const handleSelectConfidence = async (
+    index: number,
+    confidence: ConfidenceLevel
+  ) => {
+    try {
+      // since subscription.isLoading does not update when user selects answer,
+      // set isSelected to false when user selects or reselects confidence so
+      // that the loading message can display while we wait for apiClient. Then
+      // after await, set isSelected to true again
+      setSelectConfidence((prev) => ({ ...prev, isSelected: false }));
+      await apiClient.updateTeamAnswer(teamAnswerId, true, confidence);
+      setSelectConfidence((prev) => ({
+        ...prev,
+        selectedConfidenceIndex: index,
+        isSelected: true,
+      }));
+    } catch {
+      setIsConfidenceError(true);
```

Comment:
Should we set this to `false` at some point like when user retries selection?

Reply:
So currently, this is handled in the handleRetry function. The flow as of right now when a confidence-related API error occurs looks something like this: (1) `isConfidenceError` is set to true as seen above (2) `<ErrorModal>` appears as triggered by `isModalOpen={isConfidenceError}` on `line 247` (3) When the user clicks on the retry button, `handleRetry`, defined on `lines 190-203`, is called and when the condition `if (isConfidenceError)` is met, `setIsConfidenceError(false)` is called from `line 196`, setting the confidence error to false when user retries selection

Reply:
aww.nice, thanks for confirmation and the details : )